src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportProcessingTask.cs

```diff
@@ -0,0 +1,274 @@
+                }
+
+                // Initialize error store
+                IImportErrorStore importErrorStore = await _importErrorStoreFactory.InitializeAsync(GetErrorFileName(), cancellationToken);
+                result.ErrorLogLocation = importErrorStore.ErrorFileLocation;
+
+                // Load and parse resource from bulk resource
+                (Channel<ImportResource> importResourceChannel, Task loadTask) = _importResourceLoader.LoadResources(_inputData.ResourceLocation, _importProgress.CurrentIndex, _inputData.ResourceType, sequenceIdGenerator, cancellationToken);
+
+                // Import to data store
+                (Channel<ImportProcessingProgress> progressChannel, Task importTask) = _resourceBulkImporter.Import(importResourceChannel, importErrorStore, cancellationToken);
+
+                // Update progress for checkpoints
+                await foreach (ImportProcessingProgress progress in progressChannel.Reader.ReadAllAsync())
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        throw new OperationCanceledException("Import task is canceled by user.");
+                    }
+
+                    _importProgress.SucceedImportCount = progress.SucceedImportCount + succeedImportCount;
+                    _importProgress.FailedImportCount = progress.FailedImportCount + failedImportCount;
+                    _importProgress.CurrentIndex = progress.CurrentIndex;
+                    result.SucceedCount = _importProgress.SucceedImportCount;
+                    result.FailedCount = _importProgress.FailedImportCount;
+
+                    _logger.LogInformation("Import task progress: {0}", JsonConvert.SerializeObject(_importProgress));
+
+                    try
+                    {
+                        await _contextUpdater.UpdateContextAsync(JsonConvert.SerializeObject(_importProgress), cancellationToken);
+                    }
+                    catch (Exception ex)
+                    {
+                        // ignore exception for progresss update
+                        _logger.LogInformation(ex, "Failed to update context.");
+                    }
+                }
+
+                // Pop up exception during load & import
+                // Put import task before load task for resource channel full and blocking issue.
+                try
+                {
+                    await importTask;
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogInformation(ex, "Failed to import data.");
+                    throw new RetriableTaskException("Failed to import data.", ex);
```

Comment:
I'm nervous about an unknown error being labeled retriable. This could put the system in a loop if it is retrying an import that can reproduce this exception.

Reply:
So far we thought this should not happen, let me use log error for this to track if there's any unexpected error, max retry count would help to prevent it fail into infinite loop.

Reply:
I think adding a max retry count would be a good idea. I don't think it needs to be part of the first pass due to the other issues we are working through right now, but having a work item to add it would be good.

Reply:
Yes, currently we have max retry at the task level. For details: 

                TaskInfo processingTask = new TaskInfo()
                {
                    QueueId = _orchestratorInputData.ProcessingTaskQueueId,
                    TaskId = taskId,
                    TaskTypeId = ImportProcessingTask.ImportProcessingTaskId,
                    InputData = JsonConvert.SerializeObject(importTaskPayload),
                    MaxRetryCount = _orchestratorInputData.ProcessingTaskMaxRetryCount,
                };